### PR TITLE
Refactor tests to be more concise

### PR DESCRIPTION
**Pull request by matt-richardson**
The tests were very verbose, and hard to reason about.

### CODE REVIEW

File: source/OctoVersion.Tests/WhenStringifyingAnOctoVersionInfo.cs [hunk 1]
```diff
@@ -7,280 +8,120 @@ namespace OctoVersion.Tests
...
+        static readonly OctoVersionInfo VersionWithPreReleaseTag = new OctoVersionInfo(1, 2, 3, "pre", "");
+        static readonly OctoVersionInfo VersionWithPreReleaseTagAndBuildMetadata = new OctoVersionInfo(1, 2, 3, "pre", "build");
+        static readonly OctoVersionInfo VersionWithLongPreReleaseTagAndBuildMetadata = new OctoVersionInfo(2021, 1, 3, "mark-genericDocumentStore", "Branch.mark-genericDocumentStore.Sha.fb13016f3a21d7c2058fb74ab25f19e5311c6550");
+        static readonly OctoVersionInfo VersionWithLongPreReleaseTagWithSlashesAndBuildMetadata = new OctoVersionInfo(2021,1,3,"dependabot/npm_and_yarn/source/TentacleArmy.Web/ini-1.3.6","Branch.dependabot/npm_and_yarn/source/TentacleArmy.Web/ini-1.3.6.Sha.069392d9d2d37ddb6009998b92e70963badcc666");
+
+        //xunit is blurgh - IEnumerable of object[]? ick.
+        public static IEnumerable<object[]> PreReleaseTagWithDashTestCases()
+        {
+            //format is input/output/because
+            yield return new object[] { AllVersionPartsSetTo0, "", "there is no pre-release tag" };
```
Comment (cailyoung) [inline source/OctoVersion.Tests/WhenStringifyingAnOctoVersionInfo.cs:22]: Is it worth using a tuple here? Or an explicit type (`OctoVersionInfoTestData`?) defined elsewhere in this class?

**Review by cailyoung — APPROVED** [review]
Nice, much more readable without the inline data.

File: source/OctoVersion.Tests/WhenStringifyingAnOctoVersionInfo.cs (hunk 1 above)
Comment (matt-richardson) [reply source/OctoVersion.Tests/WhenStringifyingAnOctoVersionInfo.cs:22]: I'd love to use a better type - it has to be `IEnumerable<object[]>`. Not pretty.

Comment (cailyoung) [reply source/OctoVersion.Tests/WhenStringifyingAnOctoVersionInfo.cs:22]: I meant `IEnumerable<OctoVersionInfoTestData[]>`

Or it _has_ to be of type `object[]`?

Comment (matt-richardson) [reply source/OctoVersion.Tests/WhenStringifyingAnOctoVersionInfo.cs:22]: has to be an array of object 
image

**Pull request merged**